### PR TITLE
Putting Crosslink Delta Back

### PR DESCRIPTION
**Pull request by terencechain**
This PR puts cross link delta back. It was reverted for #2653. I've fixed the tests to comply with genesis slot 0

### CODE REVIEW

File: beacon-chain/core/epoch/epoch_processing_test.go [hunk 1]
```diff
@@ -966,6 +965,84 @@ func TestProcessFinalUpdates_CanProcess(t *testing.T) {
...
+		CrosslinkDataRootHash32: []byte{'A'},
+	}
+
+	rewards, penalties, err := CrosslinkDelta(state)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	attestedIndices := []uint64{1932, 500, 1790, 1015, 1477, 1211, 69}
+	for _, i := range attestedIndices {
```
Comment (shayzluf) [inline beacon-chain/core/epoch/epoch_processing_test.go]: what about checking the penalties of the indices that didn't vote

File: beacon-chain/core/epoch/epoch_processing_test.go [hunk 2]
```diff
@@ -966,6 +965,84 @@ func TestProcessFinalUpdates_CanProcess(t *testing.T) {
...
+		}
+		// Since all these validators attested, they shouldn't get penalized.
+		if penalties[i] != 0 {
+			t.Errorf("Wanted penalty balance %d, got %d",
+				0, penalties[i])
+		}
+	}
+}
+
+func TestCrosslinkDelta_CantGetWinningCrosslink(t *testing.T) {
```
Comment (shayzluf) [inline beacon-chain/core/epoch/epoch_processing_test.go]: i think we need to test rewards of this kind BaseReward(state, index) * attestingBalance / committeeBalance

Comment (shayzluf) [reply beacon-chain/core/epoch/epoch_processing_test.go]: not just BaseReward(state, index)

File: beacon-chain/core/epoch/epoch_processing_test.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/core/epoch/epoch_processing_test.go]: it's covered in `TestCrosslinkDelta_NoOneAttested`

File: beacon-chain/core/epoch/epoch_processing_test.go (hunk 2 above)
Comment (terencechain) [reply beacon-chain/core/epoch/epoch_processing_test.go]: Good point!

**Pull request merged**